### PR TITLE
Failure to add new cluster hosts

### LINKED ISSUE #809: [doc] 3  Troubleshooting cephadm
### Type of report
   - [ ] bug
   - [X] feature request

### Source Document
3  Troubleshooting cephadm:

https://documentation.suse.com/ses/7/html/ses-all/bp-troubleshooting-cephadm.html#

### Summary

> From: https://bugzilla.suse.com/show_bug.cgi?id=1179187

In case adding hosts in cephadm fail, like so:

```
$ ceph orch host add <hostname>
Failed to connect to <hostname> (<fqdn>).
  Please make sure that the host is reachable and accepts connections using the cephadm SSH key
```
Try:

1. follow the debug steps as printed in the error message

2. Add the host manually

* In case `hostname` prints a fqdn, try adding the host using the fqdn: `ceph orch host add <fqdn>`
* In case `hostname` prints a bare hostname, try adding the host using the fqdn: `ceph orch host add <hostname> <fqdn>`

If this succeeds, there might be a problem with ceph-salt.

3.  Add the host by IP:

* `ceph orch host add <hostname> <IP>`

If this succeeds, the cluster does not have proper name resolution. See: https://documentation.suse.com/ses/7/html/ses-all/deploy-cephadm.html#deploy-os

### PR DESCRIPTION
**Pull request by tbazant**
Closes #809 
Rendered version is attaached 
[bp-troubleshooting-adding-hosts_color_en.pdf](https://github.com/SUSE/doc-ses/files/5622903/bp-troubleshooting-adding-hosts_color_en.pdf)

### CODE REVIEW

File: xml/bp_troubleshooting_cephadm.xml [hunk 1]
```diff
@@ -375,17 +375,56 @@ cat config-json.json
...
-    <para>
-     Conversely, running the <command>ceph orch upgrade</command> command (see
-     <xref linkend="deploy-cephadm-day2-cephupdate-start"/>) will upgrade
-     daemons manually deployed by the PTF.
-    </para>
-   </important>
-   </sect1>
+ </sect1>
+ <sect1 xml:id="bp-troubleshooting-adding-hosts">
+  <title>Adding Hosts with &cephadm; Fails</title>
```
Comment (asettle) [inline xml/bp_troubleshooting_cephadm.xml:380]:
```suggestion
  <title>Failure When Adding Hosts with &cephadm;</title>

File: xml/bp_troubleshooting_cephadm.xml [hunk 2]
```diff
@@ -375,17 +375,56 @@ cat config-json.json
...
+ <screen>Failed to connect to <replaceable>HOSTNAME</replaceable> (<replaceable>FQDN</replaceable>).</screen>
+
+  <para>
+   For example:
+  </para>
+
+ <screen>
+    &prompt.cephuser;ceph orch host add node5.example.com
+    Failed to connect to node5 (node5.example.com).
+   </screen>
```
Comment (asettle) [inline xml/bp_troubleshooting_cephadm.xml:399]: Indentation is weird here?

**Review by asettle — COMMENTED** [review]
nits

**Review by lproven — APPROVED** [review]

File: xml/bp_troubleshooting_cephadm.xml [hunk 3]
```diff
@@ -375,17 +375,56 @@ cat config-json.json
...
-   </sect1>
+ </sect1>
+ <sect1 xml:id="bp-troubleshooting-adding-hosts">
+  <title>Adding Hosts with &cephadm; Fails</title>
+
+  <para>
+   When &cephadm; fails to add a new cluster host&mdash;no matter whether by
+   means of &cephsalt; (see <xref linkend="deploy-cephadm-configure-minions"/>)
+   or manually using the <command>ceph orch host add</command> command&mdash;you
+   can diagnose the reason of failure. The failure message has the following
```
Comment (lproven) [inline xml/bp_troubleshooting_cephadm.xml:386]:
```suggestion
   can diagnose the reason for failure. The failure message has the following
```

File: xml/bp_troubleshooting_cephadm.xml [hunk 4]
```diff
@@ -375,17 +375,56 @@ cat config-json.json
...
+  <para>
+   If the message includes a short host name (<literal>node5</literal> in our
+   example) as the <replaceable>HOSTNAME</replaceable>, try adding the host
+   using both its short name and FQDN:
+  </para>
+
+ <screen>&prompt.cephuser;ceph orch host add node5 node5.example.com</screen>
+
+  <para>
+   If any of the previous two commands succeed, &cephsalt; is probably causing
```
Comment (lproven) [inline xml/bp_troubleshooting_cephadm.xml:418]:
```suggestion
   If either of the previous two commands succeed, &cephsalt; is probably causing
```

File: xml/bp_troubleshooting_cephadm.xml [hunk 5]
```diff
@@ -375,17 +375,56 @@ cat config-json.json
...
+   the problem. Otherwise, try adding the host using its short host name and IP
+   address:
+  </para>
+
+ <screen>&prompt.cephuser;ceph orch host add node5 &wsIIIip;</screen>
+
+  <para>
+   If this succeeds, the cluster does not have a proper name resolution. Refer
```
Comment (lproven) [inline xml/bp_troubleshooting_cephadm.xml:426]:
```suggestion
   If this succeeds, the cluster does not have proper name resolution. Refer
```

**Review by lproven — CHANGES_REQUESTED** [review]

File: xml/bp_troubleshooting_cephadm.xml [hunk 6]
```diff
@@ -375,17 +375,56 @@ cat config-json.json
...
+ <screen>Failed to connect to <replaceable>HOSTNAME</replaceable> (<replaceable>FQDN</replaceable>).</screen>
+
+  <para>
+   For example:
+  </para>
+
+ <screen>
+    &prompt.cephuser;ceph orch host add node5.example.com
+    Failed to connect to node5 (node5.example.com).
+   </screen>
```
Comment (tbazant) [inline xml/bp_troubleshooting_cephadm.xml:399]:
```suggestion
<screen>
&prompt.cephuser;ceph orch host add node5.example.com
Failed to connect to node5 (node5.example.com).
</screen>
```

**Review by lproven — APPROVED** [review]

**Comment by sebastian-philipp** [comment]
The PTF change looks unrelated to this PR.

**Pull request merged**